src/Symfony/Component/Routing/Matcher/Dumper/CompiledUrlMatcherDumper.php

```diff
@@ -0,0 +1,486 @@
+ */
+class CompiledUrlMatcherDumper extends MatcherDumper
+{
+    private $expressionLanguage;
+    private $signalingException;
+
+    /**
+     * @var ExpressionFunctionProviderInterface[]
+     */
+    private $expressionLanguageProviders = array();
+
+    /**
+     * {@inheritdoc}
+     */
+    public function dump(array $options = array())
+    {
+        return <<<EOF
+<?php
+
+/**
+ * This file has been auto-generated
+ * by the Symfony Routing Component.
+ */
+
+return array(
+{$this->generateCompiledRoutes()});
+
+EOF;
+    }
+
+    public function addExpressionLanguageProvider(ExpressionFunctionProviderInterface $provider)
+    {
+        $this->expressionLanguageProviders[] = $provider;
+    }
+
+    /**
+     * Generates the arrays for CompiledUrlMatcher's constructor.
+     */
+    public function getCompiledRoutes(): array
+    {
+        // Group hosts by same-suffix, re-order when possible
+        $matchHost = false;
+        $routes = new StaticPrefixCollection();
+        foreach ($this->getRoutes()->all() as $name => $route) {
+            if ($host = $route->getHost()) {
+                $matchHost = true;
+                $host = '/'.strtr(strrev($host), '}.{', '(/)');
+            }
+
+            $routes->addRoute($host ?: '/(.*)', array($name, $route));
+        }
+
+        if ($matchHost) {
+            $compiledRoutes = array(true);
+            $routes = $routes->populateCollection(new RouteCollection());
+        } else {
+            $compiledRoutes = array(false);
+            $routes = $this->getRoutes();
+        }
+
+        list($staticRoutes, $dynamicRoutes) = $this->groupStaticRoutes($routes);
+
+        $conditions = array(null);
+        $compiledRoutes[] = $this->compileStaticRoutes($staticRoutes, $conditions);
+        $chunkLimit = \count($dynamicRoutes);
+
+        while (true) {
+            try {
+                $this->signalingException = new \RuntimeException('preg_match(): Compilation failed: regular expression is too large');
+                $compiledRoutes = array_merge($compiledRoutes, $this->compileDynamicRoutes($dynamicRoutes, $matchHost, $chunkLimit, $conditions));
+
+                break;
+            } catch (\Exception $e) {
+                if (1 < $chunkLimit && $this->signalingException === $e) {
+                    $chunkLimit = 1 + ($chunkLimit >> 1);
+                    continue;
+                }
+                throw $e;
+            }
+        }
+
+        unset($conditions[0]);
+
+        if ($conditions) {
+            foreach ($conditions as $expression => $condition) {
+                $conditions[$expression] = "case {$condition}: return {$expression};";
+            }
+
+            $checkConditionCode = <<<EOF
+    static function (\$condition, \$context, \$request) { // \$checkCondition
+        switch (\$condition) {
+{$this->indent(implode("\n", $conditions), 3)}
+        }
+    }
+EOF;
+            $checkCondition = eval('return '.$checkConditionCode.';');
```

Comment:
why do we need to eval the code here ? This code path runs even when dumping the code, and there is no reason to eval the code in such case

Reply:
fixed